### ISSUE TITLE
Reorg containing STO corrupts STODB data

### ISSUE THREAD

**Issue by zathras-crypto** [OmniLayer/omnicore#242]
While reviewing https://github.com/dexX7/bitcoin/commit/f47c8160c2656b6fcac22e5cd28b125732f470c1 I noticed that code for the ```deleteAboveBlock``` function for the STO DB actually looks like it corrupts the data when it's rewritten after a reorg.

I setup a regtest for this, output is below - check out the before and after STO DB output - as you can see the delimiter ```,``` is missing after the reorg.

```
prepare some testnet BTC
obtain an address to work with
fund the address with some testnet BTC for fees
obtain some MSC
generate some more addresses to receive the STO
send omni to addresses
send STO
send another STO
show STO DB (should be two entries)
2015-09-17 09:19:33 entry #       1= mgYT3AJ7CtkB1qq676C9fbq2hycPpk3JMs:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:585606,e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:169:1:1171214,
2015-09-17 09:19:33 entry #       2= mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:4995900747,e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:169:1:9991801493,
2015-09-17 09:19:33 entry #       3= mrWcmYEy8QuJFdV9GcpkrQgHyu67mTuPLN:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:1171216,e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:169:1:2342431,
2015-09-17 09:19:33 entry #       4= msytHgPfRLPa1Tvhe5ygvjvJDoAerV75M1:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:2342431,e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:169:1:4684862,
rolling back chain 7 blocks to trigger reorg and mining a new block
show STO DB (should be two entries)
2015-09-17 09:19:33 entry #       1= mgYT3AJ7CtkB1qq676C9fbq2hycPpk3JMs:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:585606e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:166:1:1171214,
2015-09-17 09:19:33 entry #       2= mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:4995900747e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:166:1:9991801493,
2015-09-17 09:19:33 entry #       3= mrWcmYEy8QuJFdV9GcpkrQgHyu67mTuPLN:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:1171216e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:166:1:2342431,
2015-09-17 09:19:33 entry #       4= msytHgPfRLPa1Tvhe5ygvjvJDoAerV75M1:f2f2602e2d575907510509dcc26077f7254913d35b793ab73dc6f47fe94d4d72:164:1:2342431e1b954d1734367fb69ca4c064f8cfe4dd2fde88771f62ef8d3ac9f52d50009b3:166:1:4684862,
```

Fix in the works.

**Comment by dexX7** [comment]
Uh oh, good catch!!

So just to confirm, this only affects the retrieval of STO entries, e.g. when using `omni_getsto`, but not consensus?

We have [reorg tests](https://github.com/OmniLayer/OmniJ/blob/f3522ed686c06364c1e21fca354a779b1df78d90/omnij-rpc/src/integ/groovy/foundation/omni/test/rpc/reorgs/SendToOwnersReorgSpec.groovy) for STO, but they only cover the "base consensus", i.e. the effect on balances, but not the STO recipient entries for the UI/RPC.

**Comment by zathras-crypto** [comment]
It's the DB so just 'history'.  Bad data at the UI/RPC layer in ```omni_getsto``` but no impact on consensus or balances :)

**Comment by zathras-crypto** [comment]
Fixed in https://github.com/OmniLayer/omnicore/pull/243

The same test again:
```
prepare some testnet BTC
obtain an address to work with
fund the address with some testnet BTC for fees
obtain some MSC
generate some more addresses to receive the STO
send omni to addresses
send STO
send another STO
show STO DB (should be two entries)
2015-09-17 10:40:56 entry #       1= mhpioiAuJiQPVJeGPVy6YfpVckmFNSE5ax:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:1171176,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:169:1:2342351,
2015-09-17 10:40:56 entry #       2= mmGfXCVKdcHaHsrsg9sZ96Z2bwJ2u1XnJV:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:2342351,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:169:1:4684701,
2015-09-17 10:40:56 entry #       3= mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:4995900887,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:169:1:9991801774,
2015-09-17 10:40:56 entry #       4= mtcRwMSMGh37Bb43ieFY6cwh3J7e2q5X6k:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:585586,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:169:1:1171174,
rolling back chain 7 blocks to trigger reorg and mining a new block
show STO DB (should be two entries)
2015-09-17 10:40:56 entry #       1= mhpioiAuJiQPVJeGPVy6YfpVckmFNSE5ax:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:1171176,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:166:1:2342351,
2015-09-17 10:40:56 entry #       2= mmGfXCVKdcHaHsrsg9sZ96Z2bwJ2u1XnJV:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:2342351,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:166:1:4684701,
2015-09-17 10:40:56 entry #       3= mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:4995900887,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:166:1:9991801774,
2015-09-17 10:40:56 entry #       4= mtcRwMSMGh37Bb43ieFY6cwh3J7e2q5X6k:5a9f9511fcf2e7106b0895c14ff32247390f967d465aab794e1b161c262ba949:164:1:585586,7be40ab5e8f412b0140b65f14196981fab47b9daad14d5e08fba0ef624e16abc:166:1:1171174,
```

**Comment by dexX7** [comment]
The issue was resolved via #243. :)
